File: examples/inverseqft2.cpp
sim.set_num_qubits(4);
std::vector<int> creg(4);
// Unsupported operation: reset
// Unsupported operation: reset
// Unsupported operation: reset
// Unsupported operation: reset
sim.hadamard(0, {}, {});
sim.hadamard(1, {}, {});
sim.hadamard(2, {}, {});
sim.hadamard(3, {}, {});
// Unsupported operation: barrier
sim.hadamard(0, {}, {});
creg[0] = sim.measure(0);
if (creg[0] == 1) {
    sim.gate_rz(1.5707963267948966, 1, {}, {});
}
sim.hadamard(1, {}, {});
creg[1] = sim.measure(1);
if (creg[0] == 1) {
    sim.gate_rz(0.7853981633974483, 2, {}, {});
}
if (creg[1] == 1) {
    sim.gate_rz(1.5707963267948966, 2, {}, {});
}
sim.hadamard(2, {}, {});
creg[2] = sim.measure(2);
if (creg[0] == 1) {
    sim.gate_rz(0.39269908169872414, 3, {}, {});
}
if (creg[1] == 1) {
    sim.gate_rz(0.7853981633974483, 3, {}, {});
}
if (creg[2] == 1) {
    sim.gate_rz(1.5707963267948966, 3, {}, {});
}
sim.hadamard(3, {}, {});
creg[3] = sim.measure(3);

File: examples/inverseqft2.py
import math
from qiskit import ClassicalRegister, QuantumCircuit, QuantumRegister

q = QuantumRegister(4, "q")
c = ClassicalRegister(4, "c")
qc = QuantumCircuit(q, c)

qc.reset(q)
qc.h(q)
qc.barrier(q)

qc.h(q[0])
qc.measure(q[0], c[0])
with qc.if_test((c[0], 1)):
    qc.rz(math.pi / 2, q[1])

qc.h(q[1])
qc.measure(q[1], c[1])
with qc.if_test((c[0], 1)):
    qc.rz(math.pi / 4, q[2])
with qc.if_test((c[1], 1)):
    qc.rz(math.pi / 2, q[2])

qc.h(q[2])
qc.measure(q[2], c[2])
with qc.if_test((c[0], 1)):
    qc.rz(math.pi / 8, q[3])
with qc.if_test((c[1], 1)):
    qc.rz(math.pi / 4, q[3])
with qc.if_test((c[2], 1)):
    qc.rz(math.pi / 2, q[3])

qc.h(q[3])
qc.measure(q[3], c[3])
